sos/cleaner/__init__.py

```diff
@@ -696,6 +730,19 @@ def obfuscate_report(self, archive):  # pylint: disable=too-many-branches
                     # is used. In this case we want to make sure
                     # the cleaner doesn't try to clean a binary file
                     continue
+                is_certificate = file_is_certificate(fname)
+                if is_certificate:
+                    if is_certificate == "certificatekey":
+                        # Always remove certificate Key files
+                        archive.remove_file(short_name)
+                        continue
+                    if self.opts.treat_certificates == "keep":
+                        continue
+                    if self.opts.treat_certificates == "remove":
+                        archive.remove_file(short_name)
```

Comment:
Originally, I thought we should put some debug message here "removing certificate file" or similar.

But a better approach is updating `remove_file` method in `sos/cleaner/archives/__init__.py` to have extra argument `filetype=binary` and replace:

    self.log_info(f"Removing binary file '{fname}' from archive")

to:
    self.log_info(f"Removing {filetype} file '{fname}' from archive")

(and call the method from here with proper argument)